plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostingFun/JSON.hs

```diff
@@ -0,0 +1,95 @@
+{-# LANGUAGE BangPatterns         #-}
+{-# LANGUAGE ConstraintKinds      #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE OverloadedStrings    #-}
+{-# LANGUAGE PolyKinds            #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# OPTIONS_GHC -O0               #-}
+
+-- | A separate module for JSON instances, so that we can stick @-O0@ on it and avoid spending
+-- a lot of time optimizing loads of Core whose performance doesn't matter.
+module PlutusCore.Evaluation.Machine.CostingFun.JSON () where
```

Comment:
Now in a separate file, so that we can keep `-O0` here and use the defaults for the module with `runCostingFunOneArgument` etc functions (which helped by ~0.75%, which isn't much, but also saved my sanity because I now don't need to wait an eternity to see how Core changes after a tweak).

Reply:
Getting rid of `-O0` removed a bunch of matching to extract an `Int#` from the arguments of `ModelLinearSize` etc. Which is particularly good since that matching wasn't getting cached (likely due to `-O0` as well). Now the arguments are unpacked, so we don't need to extract that `Int#` at all.